packages/flutter_test/lib/src/matchers.dart

```diff
@@ -0,0 +1,184 @@
+// Copyright 2016 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/material.dart';
+import 'package:test/test.dart';
+
+import 'finders.dart';
+
+/// Asserts that the [Finder] matches no widgets in the widget tree.
+///
+/// Example:
+///
+///     expect(find.text('Save'), findsNothing);
+const Matcher findsNothing = const _FindsWidgetMatcher(null, 0);
+
+/// Asserts that the [Finder] locates at least one widget in the widget tree.
+///
+/// Example:
+///
+///     expect(find.text('Save'), findsWidgets);
+const Matcher findsWidgets = const _FindsWidgetMatcher(1, null);
+
+/// Asserts that the [Finder] locates at exactly one widget in the widget tree.
+///
+/// Example:
+///
+///     expect(find.text('Save'), findsOneWidget);
+const Matcher findsOneWidget = const _FindsWidgetMatcher(1, 1);
+
+/// Asserts that the [Finder] locates the specified number of widgets in the widget tree.
+///
+/// Example:
+///
+///     expect(find.text('Save'), findsNWidgets(2));
+Matcher findsNWidgets(int n) => new _FindsWidgetMatcher(n, n);
```

Comment:
If the `findsWidgets` matcher (i.e. `_FindsWidgetMatcher`) were _also_ callable, then the same identifier could be reused instead of adding `N` in the name (which looks kinda weird there). Examples:

```dart
expect(find.text('Save'), findsWidgets);
expect(find.text('Save'), findsWidgets(2));
```

For this to work the `_FindsWidgetMatcher` interface would have to be along the following lines:

```dart
const CallableMatcher findsWidgets = const _FindsWidgetMatcher(1, null);

// This is needed so that callability type checks/auto-completes.
abstract class CallableMatcher {
  Matcher call(int n);
}

class _FindsWidgetMatcher extends Matcher implements CallableMatcher {
  Matcher call(int n) {
    return new _FindsWidgetMatcher(n, n);
  }
}
```